aya/src/programs/mod.rs

```diff
@@ -491,8 +499,9 @@ pub(crate) fn query<T: AsRawFd>(
     }
 }
 
-/// Detach an attached program.
+/// Detach an attached program or retrive an Fd to it
 pub trait Link: std::fmt::Debug {
+    fn fd(&mut self) -> Option<RawFd>;
```

Comment:
what is this for?

Reply:
originally I thought it was better to get the fd of a program from the `LinkRef`...
but your comments are making me think again...

removing this and making the `load` function of Extension accept `ProgramFd` would result in a better API I think...

1. I loaded the program that I wish to extend.

```rust
let mut bpf = BpfLoader::new().load_file("extension.o")?;
let prog: &mut Xdp = bpf.program_mut("main").unwrap().try_into()?;
let ext: &mut Extension = bpf.program_mut("extension").unwrap().try_into()?;
// <snip> load the main program and attach it....
ext.load(prog, "prog0")?;
ext.attach()?;
```

However, I'm not sure how to solve for the other use case:

2. Someone else (i.e another Aya program) loaded the program that I wish to extend

```rust
let mut bpf = BpfLoader::new().load_file("extension.o")?;
let prog: &mut Xdp = program_from_pinned_path("/sys/fs/bpf/prog").try_into()?; // <- What should this API be?
let ext: &mut Extension = bpf.program_mut("extension").unwrap().try_into()?;
// <snip> load the main program and attach it....
ext.load(prog, "prog0")?;
ext.attach()?;
```

Reply:
resolved based on discussion in discord. we remove this for now, and will figure out a way to support use case 2 later